src/test/java/org/apache/flink/benchmark/StreamNetworkCompressionThroughputBenchmarkExecutor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.benchmark;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.NettyShuffleEnvironmentOptions;
+import org.apache.flink.streaming.runtime.io.benchmark.StreamNetworkCompressionThroughputBenchmark;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import static org.openjdk.jmh.annotations.Scope.Thread;
+
+/**
+ * JMH throughput benchmark runner for pipelined compression scenario.
+ */
+@OperationsPerInvocation(value = StreamNetworkCompressionThroughputBenchmarkExecutor.RECORDS_PER_INVOCATION)
+public class StreamNetworkCompressionThroughputBenchmarkExecutor extends BenchmarkBase {
+
+	static final int RECORDS_PER_INVOCATION = 5_000_000;
+
+	public static void main(String[] args) throws RunnerException {
+		Options options = new OptionsBuilder()
+			.verbosity(VerboseMode.NORMAL)
+			.include(".*" + StreamNetworkCompressionThroughputBenchmarkExecutor.class.getSimpleName() + ".*")
+			.build();
+
+		new Runner(options).run();
+	}
+
+	@Benchmark
+	public void networkPipelinedCompressionThroughput(CompressionEnvironment context) throws Exception {
```

Comment:
I think this name might be too long? 

What about just `networkThroughput` with a single value parameter `COMPRESSED`? In that case it would be also semi consistent with the [existing benchmarks](http://codespeed.dak8s.net:8000/changes/)? 

Or without the parameter, but just rename to `networkThroughputCompressed`?

Or to make it fully consistent, what about just adding `"1000,100ms,COMP"` parameter to the existing benchmark
`StreamNetworkThroughputBenchmarkExecutor.MultiEnvironment#channelsFlushTimeout`?

I guess any of those three variants would be fine.

Reply:
Considering the current compression option might be experimental, then I did not plan to mix it with existing formal `StreamNetworkThroughputBenchmarkExecutor`. 
I would take the second option to only rename the method atm.